Add BaseEntity and overridden the default Django user model

umarkayondo opened the pull request:
BaseEntity is the base model from which all database models will inherit from
I have not included migrations because of the differences in different python versions. When we agree on the python version to use then we can generate the migrations for this model.

umarkayondo committed:
Add BaseEntity and overriden the default Django user model

deokyeyune commented on project/authentication/models.py:
```diff
@@ -2,7 +2,222 @@
+    """
+    ACTIVE = "Active"
+    DELETED = "Delete"
+    ACTIVE_LOCKED = "Active Locked"
```
@umarkayondo please add comments for each of these record states.

vgkienzler replied:
@umarkayondo @deokyeyune That would be a good idea indeed. I am not sure how we will take that into account/enforce it: do we count on developers to not change entries which are marked as ACTIVE_LOCKED? Or should we try to implement it in code?

deokyeyune commented on project/authentication/models.py:
```diff
@@ -2,7 +2,222 @@
+        return self.name
+
+
+class Gender(enum.Enum):
```
The complexity thats currently roaming around regarding the values Gender can hold cannot be taken lightly. Lets shift this to Lookups and Lookup classes. The Lookup classes can be enums for now but we allow users to define the lookup values. This will also save us alot of time (short cut to avoid many system settings) by transferring the responsibility to the user to define the values they want. 
@umarkayondo  and @vgkienzler

vgkienzler replied:
@deokyeyune @umarkayondo I added/changed a few things. See my [next commit ](https://github.com/vgkienzler/tickets-management-web/pull/11/commits/2dc013a6437450361f224e4c8f48b8e9d18c852b) in this branch.
In particular, in line with this comment, I have add a choice 'other' to gender. Later we can do as Deo suggests and let the user define whatever they want as gender.

vgkienzler committed:
fix: made new user work for register and logins

In details:
* Modified authentication/forms.py to use CustomUser and not django User
* Modified core/settings.py (add AUTH_USER_MODEL and DEFAULT_AUTO_FIELD)
* Modified authentication/models.py:
- replaced 'UserAccount' with 'CustomUser'
- add 'other' to Gender Enum,
- formating/commenting

vgkienzler reviewed (approved):
I added a few things to Umar's changes, in particular:
* Modified `authentication/forms.py `to use **CustomUser** and not django **User**
* Modified `core/settings.py `(add **AUTH_USER_MODEL** and **DEFAULT_AUTO_FIELD**)
* Modified `authentication/models.py`:
        - replaced `UserAccount `with `CustomUser` as name for our custom user model overrinding django's
        - add '**other**' to Gender Enum,
        - formating/commenting

vgkienzler commented:
@umarkayondo @deokyeyune I am merging this branch so that we can keep moving on. I have created an issue about making the enum 'Gender' customisable by the app user.
